Translate add-react-to-an-existing-project (#659)
reactjs/pt-br.react.dev: src/content/learn/add-react-to-an-existing-project.md

---
title: Adicionar React a um Projeto Existente
---

<Intro>

> Se você quer adicionar alguma interatividade ao seu projeto existente, você não precisa reescrevê-lo em React. Adicione React à sua pilha existente e renderize componentes React interativos em qualquer lugar.

Reviewer: A tradução de `stack` pra `pilha` acaba não fazendo muito sentido, também achei a sintaxe da segunda frase levemente confusa. 

Sugestão: 

Se você quer adicionar alguma interatividade ao seu projeto existente, você não precisa reescrevê-lo em React. Adicione React à sua aplicação e renderize componentes interativos em qualquer lugar utilizando React.

Author: Obrigado irei dar uma olhada novamente na tradução

Reviewer 2: eu usuária stack mesmo 🤔

</Intro>

<Note>

> **Você precisa instalar o [Node.js](https://nodejs.org/en/) para o desenvolvimento local.** Embora você possa [experimentar o React](/learn/installation#try-react) online ou com uma simples página HTML, realisticamente a maioria das ferramentas JavaScript que você vai querer usar para desenvolvimento requer o Node.js.

Reviewer 3: "requer" está concordando com "a maioria", está ok

> **Você precisa instalar o [Node.js](https://nodejs.org/en/) para o desenvolvimento local.** Embora você possa [experimentar o React](/learn/installation#try-react) online ou com uma simples página HTML, realisticamente a maioria das ferramentas JavaScript que você vai querer usar para desenvolvimento requer o Node.js.

Reviewer 3: ```suggestion
**Você precisa instalar o [Node.js](https://nodejs.org/pt-br/) para o desenvolvimento local.** Embora você possa [experimentar o React](/learn/installation#try-react) online ou com uma simples página HTML, realisticamente a maioria das ferramentas JavaScript que você vai querer usar para desenvolvimento requer o Node.js.
```

Reviewer 3: Acabei de ver que o glossário sugere usar links externos em pt-br quando existirem :)

</Note>

## Usando o React para uma subrota inteira do seu site existente {/*using-react-for-an-entire-subroute-of-your-existing-website*/}

Digamos que você tenha um aplicativo da web existente em `example.com` construído com outra tecnologia de servidor (como Rails) e você deseja implementar todas as rotas iniciando em `example.com/some-app/` completamente com React.

Veja como recomendamos configurá-lo:

1. **Construa a parte React do seu aplicativo** usando um dos [frameworks baseados em React](/learn/start-a-new-react-project).
2. **Especifique `/some-app` como o *caminho base*** na configuração do seu framework (veja como: [Next.js](https://nextjs.org/docs/api-reference/next.config.js/basepath), [Gatsby](https://www.gatsbyjs.com/docs/how-to/previews-deploys-hosting/path-prefix/)).
3. **Configure seu servidor ou um proxy** para que todas as solicitações em `/some-app/` sejam tratadas pelo seu aplicativo React.

Isso garante que a parte React do seu aplicativo possa [se beneficiar das melhores práticas](/learn/start-a-new-react-project#can-i-use-react-without-a-framework) embutidas nesses frameworks.

> Muitos frameworks baseados em React são full-stack e permitem que seu aplicativo React aproveite o servidor. No entanto, você pode usar a mesma abordagem mesmo se não puder ou não quiser executar JavaScript no servidor. Nesse caso, sirva a exportação HTML/CSS/JS ([saída `next export`](https://nextjs.org/docs/advanced-features/static-html-export) para o Next.js, padrão para o Gatsby) em `/some-app/` em vez disso.

Reviewer 4: ```suggestion
Muitos frameworks baseados em React são full-stack e permitem que seu aplicativo React tire proveito dos recursos do servidor. No entanto, você pode usar a mesma abordagem mesmo se não puder ou não quiser executar JavaScript no servidor. Nesse caso, exponha os HTML/CSS/JS exportados ([saída `next export`](https://nextjs.org/docs/advanced-features/static-html-export) para o Next.js, padrão para o Gatsby) em `/some-app/` em vez disso.
```

Author: Done! ✍️

> Muitos frameworks baseados em React são full-stack e permitem que seu aplicativo React tire proveito do servidor. No entanto, você pode usar a mesma abordagem mesmo se não puder ou não quiser executar JavaScript no servidor. Nesse caso, exponha os HTML/CSS/JS exportados ([saída `next export`](https://nextjs.org/docs/advanced-features/static-html-export) para o Next.js, padrão para o Gatsby) em `/some-app/` em vez disso.

Reviewer 5: ```suggestion
Muitos frameworks baseados em React são full stack e permitem que seu aplicativo React tire proveito do servidor. No entanto, você pode usar a mesma abordagem mesmo se não puder ou não quiser executar JavaScript no servidor. Nesse caso, exponha os HTML/CSS/JS exportados ([saída `next export`](https://nextjs.org/docs/advanced-features/static-html-export) para o Next.js, padrão para o Gatsby) em `/some-app/` em vez disso.
```
Full stack é usado no português sem hífen.

## Usando o React para uma parte da sua página existente {/*using-react-for-a-part-of-your-existing-page*/}

Digamos que você tenha uma página existente construída com outra tecnologia (seja uma tecnologia do servidor como Rails ou uma do cliente como Backbone) e deseje renderizar componentes React interativos em algum lugar dessa página. Essa é uma forma comum de integrar o React - na verdade, é assim que a maioria dos usos do React parecia na Meta por muitos anos!

Você pode fazer isso em duas etapas:

> 1. **Configure um ambiente JavaScript** que permita o uso da sintaxe [JSX](/learn/writing-markup-with-jsx), divida o seu código em módulos com a sintaxe [`import`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import) / [`export`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export) e utilize pacotes (por exemplo, React) do registro de pacotes [npm](https://www.npmjs.com/).

Reviewer 3: ```suggestion
1. **Configure um ambiente JavaScript** que permita o uso da sintaxe [JSX](/learn/writing-markup-with-jsx), divida o seu código em módulos com a sintaxe [`import`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/import) / [`export`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/export) e utilize pacotes (por exemplo, React) do registro de pacotes [npm](https://www.npmjs.com/).
```

<html>
  <head><title>My app</title></head>
  <body>
    <!-- O conteúdo da sua página existente (neste exemplo, ele é substituído) -->
  </body>
</html>
```

```js index.js active
import { createRoot } from 'react-dom/client';

// Limpar o conteúdo HTML existente
document.body.innerHTML = '<div id="app"></div>';

// Em vez disso, renderize seu componente React
const root = createRoot(document.getElementById('app'));
root.render(<h1>Hello, world</h1>);
```

</Sandpack>

> Se todo o conteúdo da sua página foi substituído por um "Hello World!", tudo funcionou! Continue lendo.

Reviewer 3: ```suggestion
Se todo o conteúdo da sua página foi substituído por um "Hello, world!", tudo funcionou! Continue lendo.
```
Aqui se trata do texto literal no código da linha 78

<Note>

Integrar um ambiente JavaScript modular em um projeto existente pela primeira vez pode parecer intimidante, mas vale a pena! Se você ficar preso, experimente os nossos [recursos da comunidade](/community) ou o [Vite Chat](https://chat.vitejs.dev/).

</Note>

### Passo 2: Renderizar componentes React em qualquer lugar da página {/*step-2-render-react-components-anywhere-on-the-page*/}

No passo anterior, você colocou este código no topo do seu arquivo principal:

```js
import { createRoot } from 'react-dom/client';

// Limpar o conteúdo HTML existente
document.body.innerHTML = '<div id="app"></div>';

// Em vez disso, renderize seu componente React
const root = createRoot(document.getElementById('app'));
root.render(<h1>Hello, world</h1>);
```

Claro, você não quer realmente limpar o conteúdo HTML existente!

Apague este código.

> Em vez disso, você provavelmente deseja renderizar seus componentes React em lugares específicos em seu HTML. Abra sua página HTML (ou os modelos do servidor que a geram) e adicione um atributo único [`id`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id) a qualquer tag, por exemplo:

Reviewer 4: ```suggestion
Em vez disso, você provavelmente deseja renderizar seus componentes React em lugares específicos em seu HTML. Abra sua página HTML (ou os modelos do servidor que a geram) e adicione um identificador único [`id`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id) a qualquer tag, por exemplo:
```

> Em vez disso, você provavelmente deseja renderizar seus componentes React em lugares específicos em seu HTML. Abra sua página HTML (ou os modelos do servidor que a geram) e adicione um identificador único [`id`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id) a qualquer tag, por exemplo:

Reviewer 3: ```suggestion
Em vez disso, você provavelmente deseja renderizar seus componentes React em lugares específicos em seu HTML. Abra sua página HTML (ou os modelos do servidor que a geram) e adicione um identificador único [`id`](https://developer.mozilla.org/pt-BR/docs/Web/HTML/Global_attributes/id) a qualquer tag, por exemplo:
```

</Sandpack>

> Observe como o conteúdo HTML original de index.html é preservado, mas o seu próprio componente React NavigationBar agora aparece dentro do <nav id="navigation" /> do seu HTML. Leia a documentação de uso de [`createRoot` usage documentation](/reference/react-dom/client/createRoot#rendering-a-page-partially-built-with-react) para aprender mais sobre como renderizar componentes React dentro de uma página HTML existente.

Reviewer 3: ```suggestion
Observe como o conteúdo HTML original de `index.html` é preservado, mas o seu próprio componente React `NavigationBar` agora aparece dentro do `<nav id="navigation" />` do seu HTML. Leia a documentação de uso de [`createRoot` usage documentation](/reference/react-dom/client/createRoot#rendering-a-page-partially-built-with-react) para aprender mais sobre como renderizar componentes React dentro de uma página HTML existente.
```
Suponho que os backticks devam ser mantidos?

> Quando você adota o React em um projeto existente, é comum começar com pequenos componentes interativos (como botões) e, gradualmente, "subir de nível" até que eventualmente toda a sua página seja construída com React. Se você chegar a esse ponto, recomendamos migrar para [a React framework](/learn/start-a-new-react-project) imediatamente para aproveitar ao máximo o React.

Reviewer 3: ```suggestion
Quando você adota o React em um projeto existente, é comum começar com pequenos componentes interativos (como botões) e, gradualmente, "subir de nível" até que eventualmente toda a sua página seja construída com React. Se você chegar a esse ponto, recomendamos migrar para [um framework React](/learn/start-a-new-react-project) imediatamente para aproveitar ao máximo o React.
```

> ## Usando o React Native em um aplicativo móvel nativo existente. {/*using-react-native-in-an-existing-native-mobile-app*/}

Reviewer 3: ```suggestion
## Usando o React Native em um aplicativo móvel nativo existente {/*using-react-native-in-an-existing-native-mobile-app*/}
```
O original não tem pontuação no final da frase

> [React Native](https://reactnative.dev/) também pode ser integrado incrementalmente em aplicativos nativos existentes. Se você possui um aplicativo nativo existente para Android (Java ou Kotlin) ou iOS (Objective-C ou Swift), [follow this guide](https://reactnative.dev/docs/integration-with-existing-apps) para adicionar uma tela React Native a ele.

Reviewer 2: ```suggestion
[React Native](https://reactnative.dev/) também pode ser integrado incrementalmente em aplicativos nativos existentes. Se você possui um aplicativo nativo existente para Android (Java ou Kotlin) ou iOS (Objective-C ou Swift), [siga este guia](https://reactnative.dev/docs/integration-with-existing-apps) para adicionar uma tela React Native a ele.
```
